include/deal.II/lac/sparsity_tools.h

```diff
@@ -103,6 +103,33 @@ namespace SparsityTools
                   const Partitioner partitioner = Partitioner::metis
                  );
 
+  /**
+   * Using a coloring algorithm provided by ZOLTAN to color indices whose
+   * connections are represented using a SparsityPattern object. In effect,
+   * we view this sparsity pattern as a graph of connections between nodes,
+   * where each nonzero entry in the sparsity pattern corresponds to
+   * an edge between two nodes in the connection graph. The goal is to assign
+   * each node a color index such that no two connected nodes have
+   * same color index. Note that this function requires that SparsityPattern be
+   * symmetric (and hence square) i.e a bidirected graph representation. We do
+   * not check for symmetry of the sparsity pattern, since this is an
+   * expensive operation, but rather leave this as the responsibility of
+   * caller of this function.
+   *
+   * @image html color_bidirected_graph.png
+   * For example, the image shows 5 nodes numbered from 0 to 4. The connections
+   * shown are bidirectional. After coloring, no two connected nodes have the
```

Comment:
The commonly used term, I believe, is "undirected", nor "bidirectional". You also use this in the figure itself (which I can't seem to comment on in github).